src/Spark.Mongo/Search/Searcher/CriteriaMongoExtensions.cs

```diff
@@ -399,7 +399,11 @@ private static FilterDefinition<BsonDocument> UriQuery(String parameterName, Ope
             //CK: Ugly implementation by just using existing features on the StringQuery.
             //TODO: Implement :ABOVE.
             String localModifier = "";
-            var operand2 = new UntypedValue(UriUtil.NormalizeUri(operand.ToUnescapedString()));
+            if (operand != null)
+            {
+                operand = new UntypedValue(UriUtil.NormalizeUri(operand.ToUnescapedString()));
+            }
+
```

Comment:
Remove this part
```suggestion
```

Reply:
`operand2` is only used one place in this function, see my other suggested change above. If we apply that then we do not need this block anyore.

I also see no functional change, unless I am missing something?

Reply:
Yes, operand2 is useless, so I removed it, but we can't directly use 'operand. Tounescapedstring()' because operand may be null.
![image](https://user-images.githubusercontent.com/13515020/180978051-b8304e9e-e9d1-4168-ad71-ecea29a6a1a4.png)

Reply:
I updated my suggestion above to null-check before we `new UntypedValue(...)`.

Reply:
Yes, it's good.